### PR TITLE
Collect merger statistics

### PR DESCRIPTION
**Pull request by matt-frey**
Write parcel merger statistics to the netCDF file.

### CODE REVIEW

File: src/3d/parcels/parcel_merge.f90 [hunk 1]
```diff
@@ -318,4 +332,42 @@ subroutine geometric_merge(isma, iclo, n_merge)
...
+            unique = 0
+            number = 0 ! note that "number" does not include the 'iclo' parcel in the count
+
+            !------------------------------------------------------------------
+            ! Find unique 'iclo' indices and the total number of parcels
+            ! merging with them:
+            last = 0
+            do m = 1, n_merge
+                ic = iclo(m)
+                n_big_close = n_big_close + merge(1, 0, parcels%volume(ic) > vmin)
```
Comment (sjboeing) [inline src/3d/parcels/parcel_merge.f90:353]: Does this really save over an if statement?

Comment (sjboeing) [reply src/3d/parcels/parcel_merge.f90:353]: Happy to keep this, just wondering if the compiler wouldn't take care of this.

File: src/3d/parcels/parcel_merge.f90 [hunk 2]
```diff
@@ -318,4 +332,42 @@ subroutine geometric_merge(isma, iclo, n_merge)
...
+                j = findloc(unique(1:last), value = ic, dim=1)
```
Comment (sjboeing) [inline src/3d/parcels/parcel_merge.f90:354]: Wondering if the computational cost here is unpredictable or could be high for many mergers as a findloc needs to be done for every merge. I guess the alternative would be to keep track in an array which may be oversized for what it really does?

**Review by sjboeing — COMMENTED** [review]
Just left some comments. Looks good though I was wondering if we can avoid findloc.

**Review by sjboeing — APPROVED** [review]
Happy for you to merge, Matthias. I was wondering (see slack) if there is an argument for using a temporary integer array here, rather than reusing an array, if we use temporary arrays elsewhere (and this is not the point in the code where the memory footprint is largest).

File: src/3d/parcels/parcel_merge.f90 (hunk 2 above)
Comment (matt-frey) [reply src/3d/parcels/parcel_merge.f90]: Changed.

**Pull request merged**